lib/takeaway.rb

```diff
@@ -0,0 +1,71 @@
+require_relative 'twilio'
+require_relative 'foods'
+require 'dotenv'
+Dotenv.load
+
+class Takeaway
+include Twilio
+attr_reader :ordered_dishes
+SENDER = ENV['SENDER']
+
+  def initialize
+    @menu_file = 'menu-file.txt'
+    @dish_array = []
+    @ordered_dishes = Hash.new(0)
+    @running_total = 0
+    @receiver = ENV['RECEIVER']
+  end
+
+  def menu
+    File.read(menu_file).each_line do |line|
+      puts line
+    end
+  end
+
+  def take_order(total,*dish_klasses)
+    price = dish_klasses.inject(0){|sum, k| sum + k.price}
+    if total != price
+    raise 'The total provided does not match the sum of the prices of the dishes selected'
+    end
+    @running_total += price
+    klass_iterator(dish_klasses)
+    review_order
+  end
+
+  def review_order
+    raise 'No dishes have been ordered' if @dish_array.empty?
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)